app/code/Magento/Review/Model/ReviewSummary.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Review\Model;
+
+use Magento\Framework\Model\AbstractModel;
+use Magento\Review\Model\ResourceModel\Review\Summary\CollectionFactory as SummaryCollectionFactory;
+
+/**
+ * Append review summary to product.
+ */
+class ReviewSummary
+{
+    /**
+     * @var SummaryCollectionFactory
+     */
+    private $summaryCollectionFactory;
+
+    /**
+     * @param SummaryCollectionFactory $sumColFactory
+     */
+    public function __construct(
+        SummaryCollectionFactory $sumColFactory
+    ) {
+        $this->summaryCollectionFactory = $sumColFactory;
+    }
+
+    /**
+     * Append review summary data to product
+     *
+     * @param AbstractModel $object
+     * @param int $storeId
+     * @param int $entityType
+     */
+    public function appendSummaryDataToObject(AbstractModel $object, $storeId, $entityType = 1): void
```

Comment:
Please rename `$object` and specify more accurate type for it. Other params may be `int` also.